src/FastField.tsx

```diff
@@ -307,4 +294,27 @@ class FastFieldInner<Props = {}, Values = {}> extends React.Component<
   }
 }
 
-export const FastField = connect<FieldAttributes<any>, any>(FastFieldInner);
+export namespace FastField {
+  export type Props<ExtraProps> = Field.Props<ExtraProps>;
+  export type State = {};
+}
+
+export class FastField<ExtraProps, Values> extends React.Component<
+  FastField.Props<ExtraProps>,
+  FastField.State
+> {
+  static WrappedComponent = FastFieldInner;
+
+  render() {
+    return (
+      <FormikConsumer<Values>>
+        {formik => (
+          <FastFieldInner<ExtraProps, Values>
+            {...this.props as FastField.Props<ExtraProps>}
```

Comment:
I was having trouble determining the exact need for this casting, the trace here was pretty tough to follow. My hunch was that it had to do with the lack of properly discriminating between the `GenericFieldHTMLAttributes` union combined with the  `render` / `children` / `component`. I believe it goes away if we do a better job discriminating the types earlier. Similar story in the other components.